cmd/vulcan-gitleaks/main.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2019 Adevinta
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	check "github.com/adevinta/vulcan-check-sdk"
+	"github.com/adevinta/vulcan-check-sdk/helpers"
+	checkstate "github.com/adevinta/vulcan-check-sdk/state"
+	report "github.com/adevinta/vulcan-report"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/bcrypt"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+)
+
+const (
+	DefaultDepth = 1
+)
+
+var (
+	checkName        = "vulcan-gitleaks"
+	logger           = check.NewCheckLog(checkName)
+	reportOutputFile = filepath.Join(os.TempDir(), "report.json")
+	localTargets     = []string{"localhost", "host.docker.internal", "172.17.0.1"}
+	leakedSecret     = report.Vulnerability{
+		Summary:       "Secret Leaked in Git Repository",
+		Description:   "A secret has been found stored in the Git repository. This secret may be in any historical commit and could be retrieved by anyone with read access to the repository. Test data and false positives can be marked as such.",
+		CWEID:         540,
+		Score:         float32(report.SeverityHigh),
+		ImpactDetails: "Anyone with access to the repository could retrieve the leaked secret and use it in the future with malicious intent.",
+		Labels:        []string{"issue"},
+		Recommendations: []string{
+			"Completely remove the secrets from the repository as explained in the references.",
+			"Encrypt the secrets using a tool like AWS Secrets Manager or Vault.",
+		},
+		References: []string{
+			"https://help.github.com/en/articles/removing-sensitive-data-from-a-repository",
+		},
+	}
+	details  = "This secret was found by the gitleaks rule '%s', with ID '%s'. If this doesn't correspond to a secret, it can be excluded from future scans marking it as false positive, or adding the following line to vulcan.yaml file for vulcan-local executions:\n \t- affectedResource: %s"
```

Comment:
I'm unsure about mentioning the `vulcan.yaml` here, as it isn't part of the check itself and it isn't used in regular scans. I think that kind of information should be provided by vulcan-local itself.

Reply:
I think that is a good point. I would personally keep the `vulcan.yaml` reference only if we integrated it somehow in the check itself. Otherwise this information only applies to `vulcan-local`. However, if we plan on running the check through the regular Vulcan scans I feel we should allow users to whitelist secrets proactively from a file (as with the Seekret check) and not only from the "Mark as False Positive" feature. I think it may make sense to add some ability to parse `vulcan.yaml` in the checks SDK and use that to filter those findings directly in the check. This has some implications and probably is worth it to discuss separately.

Reply:
I agree with you, this detail will be added only when scanning local resources (done here [ec07e44](https://github.com/adevinta/vulcan-checks/pull/185/commits/ec07e443a7a05c55ec762fc6ffe12642f82d8256)), since the only way of doing so for now is vulcan-local. Do you think this change is enough or would you remove the references in all of the cases?

Reply:
In my opinion this is enough. I don't know our stance in referencing Vulcan API features such as false positives on the checks, but I personally think it is fine and helpful in the current use cases. Great job 👌